pennylane/logging/configuration.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains support methods for configuring the logging functionality.
+"""
+import logging
+import logging.config
+import os
+from importlib import import_module
+from importlib.util import find_spec
+
+has_toml = False
+toml_libs = ["tomllib", "tomli", "tomlkit"]
+for pkg in toml_libs:
+    spec = find_spec(pkg)
+    if spec:
+        tomllib = import_module(pkg)
+        has_toml = True
+        break
+
+# Define absolute path to this file in source tree
+_path = os.path.dirname(__file__)
+
+# Define a more verbose mode for the messages. Not currently controlled by internal log configurations.
+TRACE = logging.DEBUG // 2
```

Comment:
Do we need this to show up in the [docs](https://xanaduai-pennylane--4383.com.readthedocs.build/en/4383/code/api/pennylane.logging.TRACE.html), it's not clear to me from the docstring what this is for.

Reply:
I guess this requires logging context (https://www.crowdstrike.com/cybersecurity-101/observability/logging-levels/, https://stackoverflow.com/questions/2031163/when-to-use-the-different-log-levels). Since Python doesn't natively have a TRACE level, this is adding that in.

This would not be something I'd expect a user to need a docstring on, as it is mentioned in the dev guide, and TOML comments. Do you prefer something different?